src/SpannerServiceProvider.php

```diff
@@ -56,40 +58,72 @@ public function boot(): void
     }
 
     /**
-     * @param array $config
+     * @param array{
+     *      name: string,
+     *      instance: string,
+     *      database: string,
+     *      prefix: string,
+     *      cache_path: string,
+     *      session_pool: array<string, mixed>,
+     * } $config
      * @return Connection
      */
     protected function createSpannerConnection(array $config): Connection
     {
-        $cache = $this->getCacheAdapter(
-            $config['name'],
-            $config['cache_path'] ?? null,
-        );
-
         return new Connection(
             $config['instance'],
             $config['database'],
             $config['prefix'],
             $config,
-            $cache,
-            new CacheSessionPool($cache, $config['session_pool'] ?? [])
+            $this->createAuthCache($config),
+            $this->createSessionPool($config),
         );
     }
 
     /**
      * @param array<string, mixed> $config
      * @param string $name
-     * @return array<string, mixed>
+     * @return array{
+     *       name: string,
+     *       instance: string,
+     *       database: string,
+     *       prefix: string,
+     *       cache_path: string,
+     *       session_pool: array<string, mixed>,
+     *  } $config
      */
     protected function parseConfig(array $config, string $name): array
     {
         return $config + [
             'prefix' => '',
             'name' => $name,
+            'cache_path' => null,
```

Comment:
```suggestion
            'cache_path' => '',
```
If it returns null, you may want to change the type hinting as well.

Reply:
type hint changed.